File: app/web/Lipspeak/backend/app/services/dashboard_service.py
"""dashboard_service.py
@authors: Anish Kumar, Bidipta Barua, Dibyasmita Hati, Arpan Haldar
@description: Dashboard business logic.
@date: 11 June 2026
@returns: Dashboard operations.
"""

# Imports
from sqlalchemy.orm import (
    Session,
)
from app.models.video import (
    Video,
)

# Dashboard Service
class DashboardService:
    """Dashboard service."""
    @staticmethod
    def get_overview(
        db: Session,
        user_id: str,
    ):
        """Get dashboard overview."""

        videos = (
            db.query(Video)
            .filter(
                Video.user_id ==
                user_id,
            )
            .order_by(
                Video.created_at.desc(),
            )
            .all()
        )

        total_videos = len(
            videos,
        )

        latest_transcript = (
            videos[0].transcript
            if videos
            else "No transcripts yet."
        )

        return {
            "total_videos":
            total_videos,

            "latest_transcript":
            latest_transcript,

            "recent_videos":
            videos[:5],
        }

File: app/web/Lipspeak/frontend/src/services/api/dashboard.service.ts
/** dashboard.service.ts
 * @authors: Anish Kumar, Bidipta Barua, Dibyasmita Hati, Arpan Haldar
 * @description: Dashboard API service.
 * @date: 11 June 2026
 * @returns: Dashboard API operations.
 *
 */


// Imports
import {
  apiClient,
} from "@/lib/api-client";


// Dashboard Service
export const dashboardService = {
  async getOverview() {
    const response =
      await apiClient.get(
        "/dashboard/overview",
      );

    return response.data;
  },
};